examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Logging;
+
+namespace WebDemoWithEventHub
+{
+    public class EventHubService
+    {
+        private IConfigurationRefresher _refresher;
+
+        private ILogger<EventHubService> _logger;
+
+        public EventHubService(IConfiguration config, IConfigurationRefresherProvider providers, ILogger<EventHubService> logger)
```

Comment:
This is an example. Let's use good naming for variables. Suggest `providers` -> `refresherProvider`.

Reply:
Updated.

---
In reply to: [464786843](https://github.com/Azure/AppConfiguration/pull/329#discussion_r464786843) [](ancestors = 464786843)

Reply:
Can we do `IOptionsSnapshot<EventHubConnection>`?

---
In reply to: [466011871](https://github.com/Azure/AppConfiguration/pull/329#discussion_r466011871) [](ancestors = 466011871,464786843)

Reply:
Changed to IOptions.

IOptionsSnapshot only works with Scoped services, and doesn't work with Singleton. And we need EventHubService to be singleton to activate it in Startup.cs.

---
In reply to: [466103302](https://github.com/Azure/AppConfiguration/pull/329#discussion_r466103302) [](ancestors = 466103302,466011871,464786843)